enterprise/backend/src/metabase_enterprise/serialization/serialize.clj

```diff
@@ -108,19 +129,74 @@
                                (u/select-non-nil-keys [:values :human_readable_values]))))))
 
+(defn- convert-column-settings-key [k]
+  (if-let [field-id (::mb.viz/field-id k)]
+    (-> (db/select-one Field :id field-id)
+        fully-qualified-name
+        mb.viz/column-ref-for-qualified-name)
+    k))
+
+(defn- convert-param-mapping-key
+  "The `k` is something like [:dimension [:fk-> [:field-id <id1>] [:field-id <id2]]]"
+  [k]
+  (mbql-id->fully-qualified-name k))
+
+(defn- convert-param-ref [new-id param-ref]
+  (cond-> param-ref
+    (= "dimension" (::mb.viz/param-ref-type param-ref)) ids->fully-qualified-names
+    (some? new-id) (update ::mb.viz/param-ref-id new-id)))
+
+(defn- convert-param-mapping-val [new-id v]
+  (-> v
+      (m/update-existing ::mb.viz/param-mapping-source (partial convert-param-ref new-id))
+      (m/update-existing ::mb.viz/param-mapping-target (partial convert-param-ref new-id))
+      (m/assoc-some ::mb.viz/param-mapping-id (or new-id (::mb.viz/param-mapping-id v)))))
+
+(defn- convert-parameter-mapping [param-mapping]
+  (if (nil? param-mapping)
+    nil
+    (reduce-kv (fn [acc k v]
+                 (assoc acc (convert-param-mapping-key k)
+                            (convert-param-mapping-val nil v))) {} param-mapping)))
+
+(defn- convert-click-behavior [{:keys [::mb.viz/link-type ::mb.viz/link-target-id] :as click}]
+  (-> (if-let [new-target-id (case link-type
+                               ::mb.viz/card      (-> (Card link-target-id)
+                                                      fully-qualified-name)
+                               ::mb.viz/dashboard (-> (Dashboard link-target-id)
+                                                      fully-qualified-name)
+                               nil)]
+        (assoc click ::mb.viz/link-target-id new-target-id)
+        click)
+      (m/update-existing ::mb.viz/parameter-mapping convert-parameter-mapping)))
+
+(defn- convert-column-settings-value [{:keys [::mb.viz/click-behavior] :as v}]
+  (cond (not-empty click-behavior) (assoc v ::mb.viz/click-behavior (convert-click-behavior click-behavior))
+        :else v))
+
+(defn- convert-column-settings [acc k v]
+  (assoc acc (convert-column-settings-key k) (convert-column-settings-value v)))
+
+(defn- convert-viz-settings [viz-settings]
+  (-> (mb.viz/from-db-form viz-settings)
+      (m/update-existing ::mb.viz/column-settings (fn [col-settings]
+                                                    (reduce-kv convert-column-settings {} col-settings)))
+      mb.viz/db-form))
+
 (defn- dashboard-cards-for-dashboard
   [dashboard]
-  (let [dashboard-cards (db/select DashboardCard :dashboard_id (u/the-id dashboard))
-        series          (when (not-empty dashboard-cards)
-                          (db/select DashboardCardSeries
-                            :dashboardcard_id [:in (map u/the-id dashboard-cards)]))]
+  (let [dashboard-cards   (db/select DashboardCard :dashboard_id (u/the-id dashboard))
```

Comment:
Not sure if this indentation change was intentional

Reply:
I think this is fixed now, or at least I reverted that spacing change in the latest.